### PR TITLE
Tiger client - expose declarativeLayout api, fix workspaces fetching

### PR DESCRIPTION
**Pull request by kandl**
- Expose declarativeLayout api
- Fix workspaces search
- Fix workspaces pagination

JIRA: RAIL-3015, RAIL-2966, RAIL-3037

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2816/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4442/ : SUCCESS in 13m 42s

File: libs/sdk-backend-tiger/src/backend/workspaces/index.ts [hunk 1]
```diff
@@ -53,41 +63,68 @@ class TigerWorkspaceQuery implements IWorkspacesQuery {
...
+}
+
+class WorkspacesInMemoryPaging extends InMemoryPaging<IAnalyticalWorkspace> {
+    constructor(
+        all: IAnalyticalWorkspace[],
+        limit = 50,
+        offset = 0,
+        public readonly search: string | undefined = undefined,
+    ) {
+        super(all, limit, offset);
```
Comment (lupko) [inline libs/sdk-backend-tiger/src/backend/workspaces/index.ts:128]: the problem here is that InMemoryPaging creates a new instance of itself on next(). the `search` field will not be available on subsequent pages.

File: libs/sdk-backend-tiger/src/backend/workspaces/index.ts [hunk 2]
```diff
@@ -53,41 +63,68 @@ class TigerWorkspaceQuery implements IWorkspacesQuery {
...
-            }),
-            limit,
-            offset,
-            totalCount: workspaces.length,
-            next: () => Promise.resolve(emptyResult),
+        const optionsToUse: OrganizationGetEntitiesOptions = {
+            query: {
+                size: limit,
+                page: offset,
+            },
```
Comment (lupko) [inline libs/sdk-backend-tiger/src/backend/workspaces/index.ts:101]: this is incorrect. page is different unit of measure than offset.

the query needs to start at page 0, load all, apply search, then start paging from the desired offset.

Comment (lupko) [reply libs/sdk-backend-tiger/src/backend/workspaces/index.ts:101]: I think you can omit these options completely. the utility will be adding its own paging..

**Review by lupko — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Review by lupko — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1125

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/828/ : SUCCESS in 1m 44s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1092/ : SUCCESS in 25s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4450/ : SUCCESS in 12m 39s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-el7/199/ : SUCCESS in 18m 03s (non-voting)

**Pull request merged**